experimental/dds/tree2/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -316,132 +324,111 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
-						}
-					} else {
-						// childChange refers to changes to node removed by some past revision. Rebase over any changes that
-						// `over` has to that same revision.
-						const overChange = overChildChanges.get(id);
-						const rebasedChild = rebaseChild(
-							childChange,
-							overChange,
-							NodeExistenceState.Dead,
-						);
-						if (rebasedChild !== undefined) {
-							perChildChanges.set(id, rebasedChild);
-						}
-					}
+					// Cell-targeting.
+					rebasedMoves.push([src, dst, target]);
 				}
+			} else {
+				// Figure out where content in src ended up in `overTagged`
+				const rebasedSrc = overSrcToDst.get(src) ?? src;
+				// Note: we cannot drop changes which map a node to itself, as this loses the intention of the original edit
+				// (since the target kind is node targeting, it may not still map to a noop after further rebases)
+				rebasedMoves.push([rebasedSrc, dst, target]);
 			}
 		}
 
-		let fieldChange: OptionalFieldChange | undefined;
-		if (change.fieldChange !== undefined) {
-			if (over.fieldChange !== undefined) {
-				const wasEmpty = over.fieldChange.newContent === undefined;
-				fieldChange = { ...change.fieldChange, wasEmpty };
-			} else {
-				fieldChange = change.fieldChange;
-			}
+		// We rebased a fill from an empty state over another edit which also sets this field.
+		// We need to make sure that we also empty the field.
+		const overFillsEmptyField = !overSrcToDst.has("self") && overDstToSrc.has("self");
+		if (overFillsEmptyField && reservedDetachId !== undefined) {
+			rebasedMoves.push(["self", reservedDetachId, "cellTargeting"]);
+			reservedDetachId = undefined;
 		}
 
-		const rebased: OptionalChangeset = {};
-		if (fieldChange !== undefined) {
-			rebased.fieldChange = fieldChange;
+		const overChildChangesBySrc = new ChildChangeMap<NodeChangeset>();
+		for (const [id, childChange] of overChange.childChanges ?? []) {
+			overChildChangesBySrc.set(overDstToSrc.get(withIntention(id)) ?? id, childChange);
 		}
-		if (perChildChanges.size > 0) {
-			rebased.childChanges = Array.from(perChildChanges.entries());
+		const rebasedChildChanges: typeof childChanges = [];
+		for (const [id, childChange] of childChanges) {
+			const rebasedId = renamedDsts.get(id) ?? id;
+			// locate corresponding child change
+			const srcId = changeDstToSrc.get(id) ?? id;
+			const overChildChange = overChildChangesBySrc.get(srcId);
+			const rebasedChildChange = rebaseChild(
+				childChange,
+				overChildChange,
+				rebasedId === "self" ? NodeExistenceState.Alive : NodeExistenceState.Dead,
+			);
+			if (rebasedChildChange !== undefined) {
+				rebasedChildChanges.push([rebasedId, rebasedChildChange]);
+			}
 		}
 
+		const overBuilds = new ChildChangeMap<boolean>();
+		for (const { id } of overChange.build) {
+			overBuilds.set(withIntention(id), true);
+		}
+		const rebased: OptionalChangeset = {
+			build: build.filter(({ id }) => !overBuilds.has(id)),
+			moves: rebasedMoves,
```

Comment:
I'm curious what motivated you drop the new build when rebasing over a build for the same ID.
Do we even expect that to happen ever? I think we don't right now, because there's no way for a branch to get the same change twice.
Once we allow merging other branches as part of transanctions then perhaps that will start occuring. When that's the case I'm still expecting that we may not want to filter.
Happy to chat about this one.

Reply:
Hah. You're correct, thanks for checking me here. There were two things going on:

1. When I added this code, I wasn't properly tagging basically any `RegisterId`s with appropriate revisions. So I'd end up with random collisions of build ids for things that don't really make sense with modular changeset's design. Interestingly, pedagogically speaking this experience made me a bit more comfortable with the interplay between change rebasers imbuing changesets with revisions and modular change family: I think that bit of the puzzle is a huge part of what I was missing while toiling away on optional field before.
2. builds weren't idempotent when I wrote this, so collision actually resulted in failure.

It's totally not necessary anymore, and is removed.